build: points dashboard wip

matthew-garrett opened the pull request:
- Adds points dashboard

jeffywu commented on apps/web/src/containers/TradingContest/points-dashboard/points-dashboard.tsx:
```diff
@@ -0,0 +1,105 @@
+                <SectionTitle sx={{ marginBottom: theme.spacing(3) }}>
+                  <FormattedMessage defaultMessage={'Season 2'} />
+                </SectionTitle>
+                <DisplayBox>April 15 - July 15</DisplayBox>
```
Can you put the seasons in a constants file and then look up the appropriate date here?

jeffywu commented on apps/web/src/containers/TradingContest/points-dashboard/points-dashboard.tsx:
```diff
@@ -0,0 +1,105 @@
+                <SectionTitle sx={{ marginBottom: theme.spacing(3) }}>
+                  <FormattedMessage defaultMessage={'Total ARB'} />
+                </SectionTitle>
+                <DisplayBox>200,000</DisplayBox>
```
Same thing with this value. Each season should have the following props:
Season Name: Season 1, Season 2, Season 3
Start Date
End Date
Total ARB
Total Points

jeffywu commented on apps/web/src/containers/TradingContest/points-dashboard/hooks/use-your-points-overview-tables.tsx:
```diff
@@ -0,0 +1,185 @@
+  const theme = useTheme();
+  const yourPointsData = [
+    {
+      activeSeason: (
```
Create a seasons config data structure for the first column and dates here.

jeffywu commented on apps/web/src/containers/TradingContest/points-dashboard/hooks/use-your-points-overview-tables.tsx:
```diff
@@ -0,0 +1,185 @@
+      ),
+      pointsPerDay: '5,120.32',
+      yourTotalPoints: '10,000.00',
+      pointsReceived: '10,000.00',
```
I don't know if the mock is wrong or not but `pointsReceived` => `arbReceived` instead.

jeffywu commented on apps/web/src/containers/TradingContest/points-dashboard/hooks/use-my-breakdown-table.tsx:
```diff
@@ -0,0 +1,125 @@
+        label: formattedTitle,
+        caption: titleWithMaturity,
+      },
+      totalAPY: 25,
```
Total APY should not be hardcoded here, it is available in the holdings data in the marketYield object.

jeffywu commented on apps/web/src/containers/TradingContest/points-dashboard/hooks/use-my-breakdown-table.tsx:
```diff
@@ -0,0 +1,125 @@
+            ? 'Open Term'
+            : `Maturity: ${formatMaturity(v.maturity)}`,
+      },
+      totalAPY: 25,
```
APY should not be hardcoded here, it is available in the vaultYield object

jeffywu commented on packages/util/src/constants.ts:
```diff
@@ -49,6 +49,27 @@ export enum Network {
   optimism = 'optimism',
 }
 
+export const PointsSeasonsData = {
```
Can you put this in the points dashboard folder instead? This is global to everything

matthew-garrett committed:
build: points dashboard wip

build: adds points dashboard sub sections

build: updates my breakdown table

build: fixes issues after rebase

build: removes console logs

build: fixes dropdown text spacing

build: hide points dashboard link

build: addresses comments

build: moves points seasons constants
